pkg/storageincentives/agent.go

```diff
@@ -323,12 +333,23 @@ func (a *Agent) play(ctx context.Context) (uint8, []byte, error) {
 
 	storageRadius := a.radius.StorageRadius()
 
+	// true if frozen
+	isFrozen, err := a.redistributionStatuser.IsOverlayFrozen(ctx)
+	if err != nil {
+		a.logger.Info("error checking if stake is frozen", "err", err)
+	}
+	a.nodeState.isFrozen(false)
+	if isFrozen {
+		a.nodeState.setLastFrozenRound(round)
+		a.nodeState.isFrozen(true)
+	}
+
 	isPlaying, err := a.contract.IsPlaying(ctx, storageRadius)
 	if !isPlaying || err != nil {
```

Comment:
we should only increment the metric on error, NOT when isPlaying is false, I know you didnt write this but can you fix it in your PR